docs/misc/components.md

```diff
@@ -0,0 +1,128 @@
+
+After configuring style to your preference,
+you can apply it to your component with either `MutableComponent#setStyle` for overwriting,
+or `#withStyle` for merging with previous one.
+
+Here's an example of how you can stylize your component:
+```java
+// Creates MutableComponent wrapping literal "Hello!"
+MutableComponent text = Component.literal("Hello!");
+
+// Copies empty style and sets color to blue and makes it italic
+Style blueItalic = Style.EMPTY
+        .withColor(0x0000FF)
+        .withItalic(true);
+
+// Copies empty style and sets color to red
+Style red = Style.EMPTY 
+        .withColor(0xFF0000);
+
+// Copies empty style and makes it bold
+Style bold = Style.EMPTY
+        .withBold(true);
+
+// Copies empty style and makes it both underlined and strikethrough
+Style doubleLines = Style.EMPTY
+        .withUnderlined(true)
+        .withStrikethrough(true);
+
+// Sets style of the text to be blue and italic
+text.setStyle(blueItalic);
+
+// Overwrites blue and italic style to be red, bold, underlined, and strikethrough
+text.withStyle(red).withStyle(bold).withStyle(doubleLines);
+```
+This results in red, bold text with two lines as shown in the [image][red_hello]
+
+Chaining Components
+-------------------
+
+`MutableComponent` can have additional components chained as siblings with `MutableComponent#append`. Chained components can be retrieved with `MutableComponent#getSiblings`.
+
+`Component` stores its siblings like a tree, and its style is merged with those of siblings as shown in the [image][tree].
+The tree is traversed in preorder.
+
+Below example will create a component with the same structure as the above image:
+```java
+// Create text only components
+MutableComponent first = Component.literal("first ");
+MutableComponent second = Component.literal("second ");
+MutableComponent third = Component.literal("third ");
+MutableComponent fourth = Component.literal("fourth ");
+MutableComponent fifth = Component.literal("fifth ");
+MutableComponent sixth = Component.literal("sixth ");
+MutableComponent seventh = Component.literal("seventh ");
+
+// Create components with style
+MutableComponent red = Component.litearl("red ").withStyle(Style.EMPTY.withColor(0xFF0000));
+MutableComponent blue = Component.literal("blue ").withStyle(Style.EMPTY.withColor(0x0000FF));
+MutableComponent bold = Component.literal("bold ").withStyle(Style.EMPTY.withBold(true));
+
+// Structure created components in the same way as the image
+red.append(first).append(blue).append(seventh);
+blue.append(second).append(third).append(bold);
+bold.append(fourth).append(fifth).append(sixth);
+```
+[Here's how it looks like in-game][style_annotated]
+
+`ComponentContents`
+-------------------
+
+Component's text contents are handled by `ComponentContents`, it holds data and defines how to represent it as text.
+Notably, subtype `TranslatableContents` not only supports localization but also text formatting.
```

Comment:
This is again unnecessary since you're just separating text formatting from the contents itself. Either more information is on the contents, or just include the first sentence as part of the general description of components.